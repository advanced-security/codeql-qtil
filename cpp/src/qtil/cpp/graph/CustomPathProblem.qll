private import qtil.locations.CustomPathProblem
private import qtil.cpp.locations.Locatable
private import cpp
// Import the C++ specific configuration for making custom path problems.
import PathProblem<Location, CppLocatableConfig>
