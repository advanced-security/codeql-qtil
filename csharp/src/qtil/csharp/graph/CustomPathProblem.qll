private import qtil.locations.CustomPathProblem
private import qtil.csharp.locations.Locatable
private import csharp
// Import the C# specific configuration for making custom path problems.
import PathProblem<Location, CsharpLocatableConfig>
