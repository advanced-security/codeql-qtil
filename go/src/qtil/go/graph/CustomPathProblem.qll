private import qtil.locations.CustomPathProblem
private import qtil.go.locations.Locatable
private import go
// Import the Go specific configuration for making custom path problems.
import PathProblem<DbLocation, GoLocatableConfig>
