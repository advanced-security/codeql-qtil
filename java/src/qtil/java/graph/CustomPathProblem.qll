private import qtil.locations.CustomPathProblem
private import qtil.java.locations.Locatable
private import java
// Import the Java specific configuration for making custom path problems.
import PathProblem<Location, JavaLocatableConfig>
