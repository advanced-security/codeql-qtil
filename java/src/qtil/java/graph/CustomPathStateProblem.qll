private import qtil.locations.CustomPathStateProblem
private import qtil.java.locations.Locatable
private import java
// Import the Java specific configuration for making custom path state problems.
import PathStateProblem<Location, JavaLocatableConfig>
