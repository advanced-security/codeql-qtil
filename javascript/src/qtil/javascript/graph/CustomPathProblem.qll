private import qtil.locations.CustomPathProblem
private import qtil.javascript.locations.Locatable
private import javascript
// Import the javascript specific configuration for making custom path problems.
import PathProblem<DbLocation, JavascriptLocatableConfig>
