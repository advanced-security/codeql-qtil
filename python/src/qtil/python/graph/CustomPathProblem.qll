private import qtil.locations.CustomPathProblem
private import qtil.python.locations.Locatable
private import python
// Import the python specific configuration for making custom path problems.
import PathProblem<Location, PythonLocatableConfig>
