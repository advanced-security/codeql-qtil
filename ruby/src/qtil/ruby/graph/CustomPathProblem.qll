private import qtil.locations.CustomPathProblem
private import qtil.ruby.locations.Locatable
private import ruby
// Import the Ruby specific configuration for making custom path problems.
import PathProblem<Location, RubyLocatableConfig>
