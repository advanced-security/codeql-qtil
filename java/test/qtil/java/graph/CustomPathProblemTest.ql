// Minimal test to make sure CustomPathProblem works with Java code.
import java
import qtil.java.graph.CustomPathProblem

module CallGraphPathProblemConfig implements CustomPathProblemConfigSig {
  class Node = Variable;

  predicate start(Node n) { n.getName() = "start" }

  predicate end(Node n) { n.getName() = "end" }

  predicate edge(Variable a, Variable b) { a.getAnAssignedValue().(VarAccess).getVariable() = b }
}

import CustomPathProblem<CallGraphPathProblemConfig>

from Variable start, Variable end
where problem(start, end)
select start, end, "Path from $@ to $@.", start.getName(), start, end.getName(), end
