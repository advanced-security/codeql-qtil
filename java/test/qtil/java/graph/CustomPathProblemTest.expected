edges
| test.java:3:9:3:20 | int mid | test.java:5:9:5:20 | int end |  |  |
| test.java:4:9:4:22 | int start | test.java:3:9:3:20 | int mid |  |  |
nodes
| test.java:3:9:3:20 | int mid | semmle.label | int mid |
| test.java:4:9:4:22 | int start | semmle.label | int start |
| test.java:5:9:5:20 | int end | semmle.label | int end |
#select
| test.java:4:9:4:22 | int start | test.java:5:9:5:20 | int end | Path from $@ to $@. | start | test.java:4:9:4:22 | int start | end | test.java:5:9:5:20 | int end |
